src/nccl/safe.rs

```diff
@@ -224,16 +237,19 @@ impl Comm {
         buff: &mut R,
         peer: i32,
     ) -> Result<result::NcclStatus, result::NcclError> {
-        unsafe {
+        buff.block_for_write(&self.stream).unwrap();
```

Comment:
CudaStream::wait() and CudaEvent::record() (which is called below) will not return `Err` unless there are previous async errors.

In order to not have to combine the API errors (NcclError + DriverError), we just call unwrap here.

May be addressed in a future PR (maybe we should combine all errors into a single mega `CudaError`